pkg/management/postgres/webserver/remote.go

```diff
@@ -142,14 +142,15 @@ func (ws *remoteWebserverEndpoints) updateInstanceManager(
 		}
 
 		// No need to do anything if we are already upgrading
-		if ws.instance.InstanceManagerIsUpgrading {
+		if !ws.instance.InstanceManagerIsUpgrading.CompareAndSwap(false, true) {
 			http.Error(w, "instance manager is already upgrading", http.StatusTeapot)
```

Comment:
I just noticed `StatusTeapot` :) Maybe 409 Conflict is more appropriate?

Reply:
Yeah, I don't have strong opinions on that!